design/accepted/gc_plus/gc-plus-milestone2-execution-plan.md

```diff
@@ -0,0 +1,50 @@
+# Uncommitted Garbage Collection - Milestone 2 Execution Plan
+
+Uncommitted Garbage Collection [Proposal](https://github.com/treeverse/lakeFS/blob/master/design/accepted/gc_plus/uncommitted-gc.md)
+
+## Milestone 1
+
+The first beta version that was released included:
+1. Implementation of the clean run flow for old and new repository structures (without optimizations)
+2. Mark & Sweep
+3. Integration tests
+4. Backup & Restore - minimal support using rclone
+
+## Milestone 2
+
+### Goals
+1. Removing the limitation of a read-only lakeFS during the GC+ job run
+
+### Non-Goals
+1. Performance improvements:
+   * [Optimaized listing on old repository structure](https://github.com/treeverse/lakeFS/issues/4620)
+   * [Efficient listing on committed entries](https://github.com/treeverse/lakeFS/issues/4600)
+   * Benchmarks - verify uncommitted GC [performance requirements](https://github.com/treeverse/lakeFS/blob/e316cafe7717bb3203e4018837a41415aa61f74b/design/accepted/gc_plus/uncommitted-gc.md?plain=1#L185) are kept
+2. [Implement optimized run flow](https://github.com/treeverse/lakeFS/issues/4489)
```

Comment:
Are we implementing the incremental run? I thought we decided to do it in the next milestone